lib/msf/base/simple/exploit.rb

```diff
@@ -66,6 +66,13 @@ def self.exploit_simple(oexploit, opts, &block)
 
       # Import options from the OptionStr or Option hash.
       exploit._import_extra_options(opts)
+      opts['Payload'] ||= exploit.datastore['Payload']
```

Comment:
If the `opt['Payload']` wasn't set, this allows it to be configured from the datastore which may have just received it on the line before where the options string may have specified it. This is the case when the user passes the payload as an argument to the `run` command (e.g. `run ... PAYLOAD=windows/x64/meterpreter/reverse_tcp`).